includes/managers/widgets.php

```diff
@@ -552,6 +552,69 @@ public function get_widgets_frontend_settings_keys() {
 		return $keys;
 	}
 
+	/**
+	 * Widgets styles names.
+	 *
+	 * Some Elementor widgets have custom styles. Each widget needs to register
+	 * the styles is depends on. Later, based on the loading strategy, Elementor
+	 * chooses how and where to enqueue these styles.
+	 *
+	 * Elementor has two types of widgets, widgets in the `/includes/widgets/`
+	 * folder and widgets in the `/modules/` folder.
+	 *
+	 * The styles are compiled to `/assets/css/widget-{$widget_name}.css` files
+	 * using grunt tasks.
+	 *
+	 * Widgets in the `/includes/widgets/` folder compile their SCSS files from
+	 * the `/assets/dev/scss/frontend/widgets/` folder.
+	 * Widgets in the `/modules/` folder compile their SCSS files from the
+	 * `/modules/{$module_name}/assets/scss/` folder.
+	 *
+	 * This method returns the list of all the widgets in the `/includes/`
+	 * folder that have styles.
+	 *
+	 * @since 3.24.0
+	 * @access public
+	 *
+	 * @return array The names of the widgets styles.
+	 */
+	public function widgets_styles_names(): array {
+		$widgets_styles = [];
+
+		// Widgets inside the `/includes/widgets/` folder that have styles.
+		$widgets = [
```

Comment:
@rami-elementor 

1. I don't know about this. It is really hard to maintain. I am thinking, the grunt process knows which widgets it created a CSS file for, maybe we should save it in a JSON - similar to `/assets/data/responsive-widgets.json`?
2. Do we need to add a hook here for 3rd parties? Hw can they leverage this loading method?

Reply:
1. We have two types of widgets, in the `/includes/widgets/` folder and in the `/modules/` folder. When using modules structure, you can register styles. Widgets in the "includes" folder can't. this is a workaround. The widget manager will register styles for them.

2. [Initially I added hooks](https://github.com/elementor/elementor/pull/28128/commits/bb628db7618100fe1a32a760bc566c0535ba6bf0) for third parties, but @KingYes prefers not to use hooks.

Reply:
@rami-elementor This method can create a JSON for the standalone widgets for example? Then we don't have to keep track of them here as well.

`https://github.com/elementor/elementor/blob/main/.grunt-config/widgets-css.js#L92-L109`

Reply:
@rami-elementor This one is important ☝️ 
(Some of the other comments were just thoughts).

Reply:
Let me see if I understand correctly, this Grunt tasks, which runs on run on build time, generate the CSS. You want the task to generate addition JSON file with all the styles in the `/includes/widgets/` folder.

The `widgets_styles_names()` method is used for registering styles in PHP. You want register CSS files, from the JSON file.

The issue is that loading file from the disk is an expensive operation. It will increase server compute, negatively impacting on TTFB. We usually try to avoid these type of solutions.

Reply:
@rami-elementor We do this for the Responsive CSS (Custom Breakpoints). I think it's here: `elementor/core/page-assets/data-managers/responsive-widgets.php`
My problem is just that when we add a new widget for example, we will never remember to come and add it here in `widgets_styles_names()`
So the Grunt task will just create this array for us (in JSON format), instead of us trying to maintain it manually.

Reply:
new widgets are added only using modules, not in the includes folder.